// Copyright 2016 The Cartographer Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package cartographer.mapping.proto;

message GncOptions2D {
  // Enable GNC Matching
  bool use_gnc = 1;
  // Number of max iterations before solver abort is returned by gnc
  int32 max_iterations = 2;
  // When this value is reached, the optimization is completed
  double non_convexity_stop = 3;
  // Shape of the GM Function
  double gm_shape = 4;
  // Minimal start non convexity
  double min_convexity = 5;
  // Factor used by non convexity increment: non_convexity = non_convexity / non_convexity_inc_factor
  double non_convexity_inc_factor = 6;
  // If non convexity fails, it repeats this number of times with different gm-shapes
  int32 max_retries = 7;
}
